quickwit/quickwit-search/src/leaf.rs

```diff
@@ -976,6 +1080,8 @@ async fn leaf_search_single_split_wrapper(
         }),
     }
     if let Some(last_hit) = locked_incremental_merge_collector.peek_worst_hit() {
+        // TODO: we could use a RWLock instead and read the value instead of updateing it
+        // unconditionally.
```

Comment:
does it matter? we only update it once per split don't we?

Reply:
I saw some contention on `incremental_merge_collector` when pushing results, which is also once per split.

It's more an issue when we start to share the threshold between indices search (which we don't currently)